commit fbd04e96ccb9458bcf612b4f1b4337f34a092e19
Author: Avijit Kanti Das <avijitnsec@codeaurora.org>
Date:   Thu Jun 8 06:41:15 2017 -0700

    UBUNTU: SAUCE: (no-up) net: Zeroing the structure ethtool_wolinfo in ethtool_get_wol()
    
    CVE-2014-9900
    
    memset() the structure ethtool_wolinfo that has padded bytes
    but the padded bytes have not been zeroed out.
    
    Change-Id: If3fd2d872a1b1ab9521d937b86a29fc468a8bbfe
    Signed-off-by: Avijit Kanti Das <avijitnsec@codeaurora.org>
    Signed-off-by: Brad Figg <brad.figg@canonical.com>
    Acked-by: Seth Forshee <seth.forshee@canonical.com>
    Acked-by: Colin Ian King <colin.king@canonical.com>
    Signed-off-by: Seth Forshee <seth.forshee@canonical.com>

diff --git a/net/core/ethtool.c b/net/core/ethtool.c
index 9a9a3d77e327..fbdf65c315a6 100644
--- a/net/core/ethtool.c
+++ b/net/core/ethtool.c
@@ -1450,11 +1450,13 @@ static int ethtool_reset(struct net_device *dev, char __user *useraddr)
 
 static int ethtool_get_wol(struct net_device *dev, char __user *useraddr)
 {
-	struct ethtool_wolinfo wol = { .cmd = ETHTOOL_GWOL };
+	struct ethtool_wolinfo wol;
 
 	if (!dev->ethtool_ops->get_wol)
 		return -EOPNOTSUPP;
 
+	memset(&wol, 0, sizeof(struct ethtool_wolinfo));
+	wol.cmd = ETHTOOL_GWOL;
 	dev->ethtool_ops->get_wol(dev, &wol);
 
 	if (copy_to_user(useraddr, &wol, sizeof(wol)))
